src/storage-client/src/statistics.rs

```diff
@@ -296,12 +296,33 @@ impl From<bool> for Boolean {
 /// A numerical gauge that never regresses.
 ///
 /// Defaults to 0, and can be left unnitialized.
-#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Default)]
-pub struct Total(Option<u64>);
+///
+/// If a metric is provided, it is bumped on regressions, as opposed to `error!`
+/// logs.
```

Comment:
Feels like this semantically would be clearer with named struct fields
```
pub struct Total(
    count: Option<u64>,
    regressions: Option<
        mz_ore::metrics::DeleteOnDropCounter<'static, prometheus::core::AtomicU64, Vec<String>>,
    >
}
```

Reply:
good point!